dask/array/optimization.py

```diff
@@ -37,6 +60,129 @@ def optimize(dsk, keys, **kwargs):
 
     dsk = convert_legacy_graph(dsk)
     dsk = fuse_linear_task_spec(dsk, keys=keys)
+    dsk = optimize_slices(dsk)
+
+    return dsk
+
+
+def _is_getter_task(
+    value: GraphNode,
+) -> tuple[Callable, Any, Any, bool, bool | None] | None:
+    """Check if a value in a Dask graph looks like a getter.
+    1. Is it a tuple with the first element a known getter.
+    If a getter is found, it returns a tuple with (getter, array, index, asarray, lock).
+    Otherwise it returns ``None``.
+    """
+    if not isinstance(value, Task):
+        return None
+    func = value.func
+    if func in GETTERS:
+        get = func
+    else:
+        return None
+
+    length = len(value.args)
+    if length == 2:
+        # getter defaults to asarray=True, getitem is semantically False
+        return get, value.args[0], value.args[1], get is not getitem, None
+    elif length == 4:
+        return get, *list(value.args)
+
+    return None
+
+
+def optimize_slices(dsk):
+    """Optimize slices
+    1.  Fuse repeated slices, like x[5:][2:6] -> x[7:11]
+
+    This is generally not very important since we are fusing those tasks anyway. There
+    is one specific exception to how xarray implements opening netcdf files and subsequent
+    slices. Not merging them together can cause reading the whole netcdf file before
+    we drop the unnecessary data. Fusing slices avoids that pattern.
+
+    See https://github.com/pydata/xarray/issues/9926
```

Comment:
Can we somehow confirm this is actually working now? A manual confirmation is good enough for me at this point

Reply:
I tested this manually with netcdf files, but the test also validates this behavior more generally